rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java

```diff
@@ -1055,8 +971,61 @@ private void internalInitializePersistedInfraVariables() {
      */
     protected abstract void initializePersistedInfraVariables();
 
-    public void setPersistedNodeSourceData(NodeSourceData nodeSourceData) {
-        this.nodeSourceData = nodeSourceData;
+    private void internalInitializePersistedInfraVariables() {
+        persistedInfraVariables.put(DEPLOYING_NODES_URL_KEY, new HashSet<>());
+        persistedInfraVariables.put(LOST_NODES_URL_KEY, new HashSet<>());
+        persistedInfraVariables.put(ACQUIRED_NODES_NAME_KEY, new HashSet<>());
+        persistedInfraVariables.put(USING_DEPLOYING_NODES_KEY, false);
+        persistedInfraVariables.put(RM_URL_KEY, "");
+    }
+
+    /**
+     * Add the information of the given deploying node to the database.
+     *
+     * @param rmNode the node to add to the database
+     */
+    private void persistNewRMDeployingNodeIfRecoveryEnabled(RMDeployingNode rmNode) {
+        createDBManagerIfNull();
+        if (nodeRecoveryEnabled()) {
+            RMNodeData rmNodeData = RMNodeData.createRMNodeData(rmNode);
+            NodeSourceData nodeSourceData = dbManager.getNodeSource(rmNode.getNodeSourceName());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "nodeSourceData" which hides the field declared at line 143. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)